### PR TITLE
speed up git-find-large files with binary search

### PR DESCRIPTION
**Pull request by larsxschneider**
Sort the temporary `git rev-list` results and use `look` to read the
result using binary sort.

#### Example benchmark

On a repo with >5 million objects finding the hash is ~1000 faster:

```
time grep 6d2793720d350abdfbf91a286f427429550edc8c unsorted
6d2793720d350abdfbf91a286f427429550edc8c some/file
grep --color=auto --exclude-dir={.bzr,CVS,.git,.hg,.svn}  test  5.49s user 0.06s system 99% cpu 5.555 total
```
```
time look 6d2793720d350abdfbf91a286f427429550edc8c sorted
6d2793720d350abdfbf91a286f427429550edc8c some/file
look 6d2793720d350abdfbf91a286f427429550edc8c sorted  0.00s user 0.00s system 86% cpu 0.005 total
```

### CODE REVIEW

**Comment by larsxschneider** [comment]
@loverso based on your work, this should make it even faster 😄

**Comment by loverso** [comment]
TL;DR: On Linux you need to use "look -b" not "look"

I ran this multiple times and found the version using 'look' to be significantly slower than the version using 'grep'.

With grep, just around 2 hours.
$ time ../../bin/../git-repo-analysis.grep/git-find-large-files > /tmp/flf.g
1914.288u 6910.668s 2:15:15.62 108.7%   0+0k 21742360+2284536io 143

With look, over 3 hours
$ time ../../bin/../git-repo-analysis.look/git-find-large-files > /tmp/flf.l
6641.952u 5221.920s 3:07:02.10 105.7%   0+0k 11299752+4930880io 1434998pf+0w
6245pf+0w

Micro-benchmark, grep is faster than look for me:

$ time look f950fd49449861768852633fd5db40ed5b0ea692 /tmp/git-find-large-files.Ztwa5e/objects > /dev/null
0.424u 0.040s 0:00.46 100.0%    0+0k 0+0io 0pf+0w

$ time grep f950fd49449861768852633fd5db40ed5b0ea692 /tmp/git-find-large-files.Ztwa5e/objects > /dev/null
0.092u 0.096s 0:00.19 94.7%     0+0k 0+0io 0pf+0w

FWIW, /tmp is an SSD on my machine, but I've got 64GB of memory, so this should all be cached in anyway.  I'm running Debian 9.

I noticed you sorted before creating the temp file, which is necessary for 'look' to be faster.  But then I caught this in the Debian man page https://manpages.debian.org/stretch/bsdmainutils/look.1.en.html:

     -b     Use a binary search on the given word list. ... Please note that these
             options are the default if no filename is given.

With '-b', it is much faster (and gives correct output):

$ time look -b f950fd49449861768852633fd5db40ed5b0ea692 /tmp/git-find-large-files.Ztwa5e/objects > /dev/null
0.000u 0.000s 0:00.00 0.0%      0+0k 0+0io 0pf+0w


MacOS and FreeBSD do not have the '-b' option and they correctly default to binary search. Similarly, the 'look' from the util-linux package works the same way, and thus CentOS and RedHat are correct.  However, Ubuntu seems to use the same 'look' as Debian and thus requires '-b'.

So I think you either need logic to detect and supply '-b' or just keep using grep.

**Comment by loverso** [comment]
Overall runtime with 'look -b' is faster by an hour:

$ time ../../bin/../git-repo-analysis/git-find-large-files > /tmp/flf.lb
82.464u 4952.028s 1:15:46.90 110.7%     0+0k 980448+4927816io 1078378pf+0w
